crates/miden-objects/src/account/component/mod.rs

```diff
@@ -195,11 +195,8 @@ impl AccountComponent {
 
     /// Returns the digest of the procedure with the specified name, or `None` if it was not found
     /// in this component's library or its library path is malformed.
-    pub fn get_procedure_root_by_name(
-        &self,
-        proc_name: impl TryInto<QualifiedProcedureName>,
-    ) -> Option<Word> {
-        self.code.as_library().get_procedure_root_by_name(proc_name)
+    pub fn get_procedure_root_by_name(&self, proc_name: impl AsRef<Path>) -> Option<Word> {
```

Comment:
I think we should probably rename this to keep things consistent.

Ideally, we'd actually be able to find procedures by name, but for this we'd need to make sure all procedures in a component are uniquely named - and this will probably require some re-work. Let's create an issue for this.